### PR TITLE
Implemented VJ Program base and made updates to NFBServer

### LINKED ISSUE #206: Implement VJ Program Mode
**Actual Behaviour.**  
The current VJ Program mode is a blank fragment. The base and the Activity needs to be coded and integrated.

**Expected Behaviour**  
Presence of the VJ Program Mode Activity on clicking the VJ button from the launcher.

**Would you like to work on the issue?**
Yes, I would work on this.

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #206 

**Changes**: Implemented VJ Program base and made updates to NFBServer

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

### CODE REVIEW

**Comment by jddeep** [comment]
It is a WIP. Need to add #208 #209 #210 firstly before this.

File: app/src/main/java/io/neurolab/program_modes/VJ.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package io.neurolab.program_modes;
+
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/program_modes/VJ.java:2]: @jddeep you can remove blank spaces from here.

File: app/src/main/java/io/neurolab/program_modes/VJ.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    private Context context;
+    private NFBServer rn;
+    private FeedbackSettings currentFeedbackSettings;
+
+    public void run(Context context) {
+        this.context = context;
+        Config config = new Config("filename");
+        try {
+
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/program_modes/VJ.java:30]: @jddeep also here.

File: app/src/main/java/io/neurolab/program_modes/VJ.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            tasks.add(fftPreprocessor);
+            tasks.add(nfbProcessor);
+            tasks.add(longtermFFTVisualizer);
+            tasks.add(oscForwardTask);
+            rn.setTasks(tasks);
+
+            rn.setCurrentFeedbackSettings(currentFeedbackSettings);
+
+            addFeedback();
+
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/program_modes/VJ.java:56]: @jddeep here as well

**Comment by suresh-gandhi** [comment]
@jddeep Good work. Rest everything is fine and good to go. Are you going to update the PR after 208, 209 and 210 get merged (as they have cross dependencies upon each other)? Also, use the reformat and rearrange code options in Android Studio as they would save you from a ton of headache of doing it manually and precisely at every line.

File: app/src/main/java/io/neurolab/program_modes/VJ.java (hunk 1 above)
Comment (jddeep) [reply app/src/main/java/io/neurolab/program_modes/VJ.java]: > @jddeep Good work. Rest everything is fine and good to go. Are you going to update the PR after 208, 209 and 210 get merged (as they have cross dependencies upon each other)? Also, use the reformat and rearrange code options in Android Studio as they would save you from a ton of headache of doing it manually and precisely at every line.

Yes @suresh-gandhi I need to work on #208 #209 @210 first before getting this merged.

**Comment by jddeep** [comment]
> @jddeep Good work. Rest everything is fine and good to go. Are you going to update the PR after 208, 209 and 210 get merged (as they have cross dependencies upon each other)? Also, use the reformat and rearrange code options in Android Studio as they would save you from a ton of headache of doing it manually and precisely at every line.

Yes @suresh-gandhi I need to work on #208 #209 #210 first before getting this merged.

**Comment by mariobehling** [comment]
Please resolve conflicts.

**Review by mariobehling — APPROVED** [review]

**Pull request merged**